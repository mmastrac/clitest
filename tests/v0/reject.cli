#!/usr/bin/env clitest --v0

reject {
    ! Hello, world!
    sequence {
        ! a
        ! b
        ! c
    }
}


# if true {
#     # Rejects are global, this should escape the if block
#     reject {
#         ! Something else
#     }
# }

$ printf "Hello, world!\n"
%EXPECT_FAILURE
*

# This doesn't work yet because we don't treat preamble patterns as commands.

# $ printf "Something else\n"
# %EXPECT_FAILURE
# *

$ printf "a\nb\nnot c\n"
*

$ printf "a\nb\nc\n"
%EXPECT_FAILURE
*

$ printf "fails"
%EXPECT_FAILURE
reject {
    ! fails
}
*

# The previous rejection was scoped to that command
$ printf "fails"
! fails
