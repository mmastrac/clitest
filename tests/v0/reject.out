Running tests/v0/reject.cli ...

printf "Hello, world!\n"
---
Hello, world!
---
âœ… OK

tests/v0/reject.cli PASSED
1 test(s) passed
